### PR TITLE
Added release workflow

### LINKED ISSUE #43: GLIBC version error on Debian
Trying to run binaries from the release page on Debian output the following error.
```
./drg-linux-amd64: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./drg-linux-amd64)
```

### PR DESCRIPTION
**Pull request by vedangj044**
Fixes #43 

This GitHub action workflow runs at the time of release and uploads a `drg-linux` binary file compiled on **Ubuntu** to the release assets.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+on:
+  release:
+    branches:
+      - main
+    types: [created]
+name: Publish release
+jobs:
+  release_linux:
+    runs-on: ubuntu-latest
```
Comment (ctron) [inline .github/workflows/release.yml:9]: "latest" is problematic, as that could change at any time and re-create the issue.

You need to pick a version which has a stable GLIBC. You can also run this inside a container:

https://github.com/ctron/hat/blob/444f4ad1f3c5cdb2e309fe50d1812676ea715ddc/.travis.yml#L19

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+on:
+  release:
+    branches:
+      - main
+    types: [created]
+name: Publish release
+jobs:
+  release_linux:
```
Comment (ctron) [inline .github/workflows/release.yml:8]: I would suggest a build matrix instead: https://github.com/ctron/hat/blob/master/.travis.yml#L9

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release
+      - name: Rename
+        run: mv ./target/release/drg ./target/release/drg-linux
+      - uses: AButler/upload-release-assets@v2.0
```
Comment (ctron) [inline .github/workflows/release.yml:25]: You can use the "hub" tool to do the same:

https://github.com/drogue-iot/drogue-cloud/blob/77c264c5bd19bf0005194de44e828d23b1594dd1/.github/workflows/ci.yaml#L255-L276

**Review by ctron — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 3 above)
Comment (jbtrystram) [reply .github/workflows/release.yml:25]: there are some actions as well https://github.com/softprops/action-gh-release

File: .github/workflows/release.yml (hunk 1 above)
Comment (vedangj044) [reply .github/workflows/release.yml]: Changed to specific tag Ubuntu-18.04

Works on
- Centos:8
- Red hat universal base image
- Fedora 34
- Arch Linux
- Debian ( after `apt-get install libssl-dev` )

File: .github/workflows/release.yml (hunk 3 above)
Comment (vedangj044) [reply .github/workflows/release.yml]: I made changes, now we use `hub` tool

Previously, I used this action softprops/action-gh-release but it has some issues which need a workaround, thus switched to other methods.

File: .github/workflows/release.yml (hunk 2 above)
Comment (vedangj044) [reply .github/workflows/release.yml:8]: I think we can add the matrix tag when we work on cross-compilation.

File: .github/workflows/release.yml (hunk 1 above)
Comment (jbtrystram) [reply .github/workflows/release.yml]: For posterity: Debian users can install `apt-get install libssl1.1` if they don't want a `dev` dependency

**Pull request merged**